pkg/datastore/storeadapter/applicationstore.go

```diff
@@ -0,0 +1,218 @@
+package storeadapter
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/oam-dev/kubevela/apis/core.oam.dev/v1beta1"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/oam-dev/kubevela/pkg/utils/apply"
+
+	"github.com/oam-dev/velacp/pkg/datastore"
+	"github.com/oam-dev/velacp/pkg/proto/model"
+	"github.com/oam-dev/velacp/pkg/runtime"
+	kruntime "k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	applicationKind = "application"
+)
+
+var _ ApplicationStore = &applicationStore{}
+
+type ApplicationStore interface {
+	AddApplication(application *model.Application, clusterName string) error
+	PutApplication(application *model.Application, clusterName string) error
+	ListApplications(appName string) ([]*model.Application, error)
+	DelApplication(appName, clusterName string) error
+}
+
+func NewApplicationStore(ds datastore.DataStore) ApplicationStore {
+	return &applicationStore{
+		ds: ds,
+	}
+}
+
+type applicationStore struct {
+	ds datastore.DataStore
+}
+
+func (c *applicationStore) AddApplication(application *model.Application, clusterName string) error {
+	ctx := context.Background()
+	var cluster model.Cluster
+	if err := c.ds.Get(ctx, clusterKind, clusterName, &cluster); err != nil {
```

Comment:
Since we have added ClusterStore in #94 , it is better to move the cluster query logic into AppService. AppService can have both ClusterStore and AppStore